Modix.Services/UserInfo/UserInfoCommand.cs

```diff
@@ -43,55 +37,41 @@ public class UserInfoCommandHandler : AsyncRequestHandler<UserInfoCommand>
         private readonly IDiscordClient _discordClient;
         private readonly IAutoRemoveMessageService _autoRemoveMessageService;
         private readonly IImageService _imageService;
+        private readonly ILogger<UserInfoCommandHandler> _logger;
+        private readonly Stopwatch _stopwatch = new Stopwatch();
 
         public UserInfoCommandHandler(ModixContext modixContext, IDiscordClient discordClient,
-            IAutoRemoveMessageService autoRemoveMessageService, IImageService imageService)
+            IAutoRemoveMessageService autoRemoveMessageService, IImageService imageService, ILogger<UserInfoCommandHandler> logger)
         {
             _modixContext = modixContext;
             _discordClient = discordClient;
             _autoRemoveMessageService = autoRemoveMessageService;
             _imageService = imageService;
+            _logger = logger;
         }
 
-        protected override async Task Handle(UserInfoCommand request,
-            CancellationToken cancellationToken)
+        protected override async Task Handle(UserInfoCommand request, CancellationToken cancellationToken)
         {
+            _stopwatch.Start();
+
+            _logger.LogDebug("Getting info for user {UserId} in guild {GuildId}", request.UserId, request.GuildId);
+
             var guild = await _discordClient.GetGuildAsync(request.GuildId);
-            var discordUser = await _discordClient.GetUserAsync(request.UserId);
 
-            var modixUser = await _modixContext
-                .GuildUsers
-                .Where(x => x.UserId == request.UserId)
-                .Where(x => x.GuildId == request.GuildId)
-                .Select(x => new
-                {
-                    x.FirstSeen,
-                    x.LastSeen,
-                    Infractions = x.Infractions
-                        .Where(d => d.DeleteActionId == null)
-                        .Select(d => new
-                        {
-                            Type = d.Type,
-                            Date = d.CreateAction.Created,
-                            Reason = d.Reason,
-                        }).ToList(),
-                    Promotions = x.PromotionCampaigns
-                        .Where(d => d.Outcome == PromotionCampaignOutcome.Accepted)
-                        .Where(d => d.CloseActionId != null)
-                        .Select(d => new
-                        {
-                            Date = d.CloseAction!.Created,
-                            TargetRoleId = d.TargetRoleId,
-                        }).ToList(),
-                })
-                .FirstOrDefaultAsync();
+            var discordUser = await guild.GetUserAsync(request.UserId);
+
+            _logger.LogDebug("Got user from Discord API");
+
+            var modixUser = await GetUserAsync(request.UserId, request.GuildId, cancellationToken);
 
             if (discordUser is null
-                && modixUser is null)
+                || modixUser is null)
```

Comment:
I'm thinking that `modixUser` could be null if we're searching for a user who exists but was never guild member (for an example use case, I've often seen people run `!info` on users who DM them out of nowhere). I'd like to make sure we aren't losing that capability here

Reply:
I believe I will need a separate branch for this scenario, I intentionally limited it to guild users for the benefit of one less API call. If the user doesn't exist in the MODiX database, I'll make a separate branch to query the user from the Discord API only then.